src/main/scala/it/pagopa/interop/apigateway/service/impl/NotifierServiceImpl.scala

```diff
@@ -0,0 +1,48 @@
+package it.pagopa.interop.apigateway.service.impl
+
+import it.pagopa.interop.apigateway.service.{NotifierInvoker, NotifierService}
+import it.pagopa.interop.commons.utils.errors.GenericComponentErrors
+import it.pagopa.interop.notifier.client.api.EventsApi
+import it.pagopa.interop.notifier.client.invoker.BearerToken
+import it.pagopa.interop.notifier.client.model.Messages
+import it.pagopa.interop.partymanagement.client.invoker.ApiError
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class NotifierServiceImpl(invoker: NotifierInvoker, api: EventsApi)(implicit ec: ExecutionContext)
+    extends NotifierService {
+
+  implicit val logger: Logger = LoggerFactory.getLogger(this.getClass)
+
+  override def getEvents(lastEventId: String, limit: Int)(contexts: Seq[(String, String)]): Future[Messages] = {
+    for {
+      (bearerToken, correlationId, ip) <- extractHeadersWithOptionalCorrelationIdF(contexts)
+      request = api.getEventsFromId(
+        lastEventId,
+        xCorrelationId = Some(correlationId),
+        xForwardedFor = ip,
+        limit = Some(limit)
+      )(BearerToken(bearerToken))
+      result <- invoker.invoke(
+        request,
+        "Retrieving message events",
+        handleCommonErrors(s"Error while retrieving events from $lastEventId")
```

Comment:
if suggestion on different pagination will be accepted, this could not be necessary anymore